### PR TITLE
Add --force support for Http Target Proxy; and fix little bug in forwarding rule syncer

### PR DESCRIPTION
**Pull request by G-Harmon**
cc @nikhiljindal @csbell @madhusudancs

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/k8s-multicluster-ingress/76)

### CODE REVIEW

File: app/kubemci/pkg/gcp/forwardingrule/forwardingrulesyncer.go [hunk 1]
```diff
@@ -125,20 +125,22 @@ func (s *ForwardingRuleSyncer) GetLoadBalancerStatus(lbName string) (*status.Loa
...
-	fmt.Println("Updating existing forwarding rule", name, "to match the desired state")
 	// We do not have an UpdateForwardingRule method.
 	// If target proxy link is the only thing that is different, then we can call SetProxyForGlobalForwardingRule.
 	// Else, we need to delete the existing rule and create a new one.
-	if existingFR.IPAddress != desiredFR.IPAddress || existingFR.PortRange != desiredFR.PortRange || existingFR.IPProtocol != desiredFR.IPProtocol {
+	if existingFR.IPAddress != desiredFR.IPAddress || existingFR.PortRange != desiredFR.PortRange ||
+		existingFR.IPProtocol != desiredFR.IPProtocol || existingFR.Description != desiredFR.Description {
 		fmt.Println("Deleting the existing forwarding rule", name, "and will create a new one")
 		if err := utils.IgnoreHTTPNotFound(s.frp.DeleteGlobalForwardingRule(name)); err != nil {
+			fmt.Println("Error deleting global forwarding rule.")
```
[the remark this answers is not in the document]
Comment (G-Harmon) [reply app/kubemci/pkg/gcp/forwardingrule/forwardingrulesyncer.go]: Done. I had added all these lines when I was trying to figure out *which* step was failing. (The error message itself showed up fine at the end of the invocation.)

File: app/kubemci/pkg/gcp/forwardingrule/forwardingrulesyncer.go [hunk 2]
```diff
@@ -125,20 +125,22 @@ func (s *ForwardingRuleSyncer) GetLoadBalancerStatus(lbName string) (*status.Loa
...
 			return fmt.Errorf("error in deleting existing forwarding rule %s: %s", name, err)
 		}
 		return s.createForwardingRule(desiredFR)
 	}
 	// Update target proxy link in forwarding rule.
 	err := s.frp.SetProxyForGlobalForwardingRule(name, desiredFR.Target)
 	if err != nil {
+		fmt.Println("Error setting proxy for forwarding rule. Target:", desiredFR.Target)
```
[the remark this answers is not in the document]
Comment (G-Harmon) [reply app/kubemci/pkg/gcp/forwardingrule/forwardingrulesyncer.go]: Done.

File: app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer.go [hunk 3]
```diff
@@ -142,24 +142,28 @@ func (l *LoadBalancerSyncer) CreateLoadBalancer(ing *v1beta1.Ingress, forceUpdat
 	ingAnnotations := annotations.IngAnnotations(ing.ObjectMeta.Annotations)
 	// Configure HTTP target proxy and forwarding rule, if required.
 	if ingAnnotations.AllowHTTP() {
-		tpLink, tpErr := l.tps.EnsureHttpTargetProxy(l.lbName, umLink)
+		tpLink, tpErr := l.tps.EnsureHttpTargetProxy(l.lbName, umLink, forceUpdate)
 		if tpErr != nil {
 			// Aggregate errors and return all at the end.
+			fmt.Println("Error ensuring HTTP target proxy.")
```
[the remark this answers is not in the document]
Comment (G-Harmon) [reply app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer.go]: Done.

File: app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer.go [hunk 4]
```diff
@@ -142,24 +142,28 @@ func (l *LoadBalancerSyncer) CreateLoadBalancer(ing *v1beta1.Ingress, forceUpdat
...
 			err = multierror.Append(err, tpErr)
 		}
 		frErr := l.frs.EnsureHttpForwardingRule(l.lbName, ipAddr, tpLink, clusters, forceUpdate)
 		if frErr != nil {
 			// Aggregate errors and return all at the end.
+			fmt.Println("Error ensuing http forwarding rule.")
```
[the remark this answers is not in the document]
Comment (G-Harmon) [reply app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer.go]: Done.

File: app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer.go [hunk 5]
```diff
@@ -142,24 +142,28 @@ func (l *LoadBalancerSyncer) CreateLoadBalancer(ing *v1beta1.Ingress, forceUpdat
...
 			err = multierror.Append(err, frErr)
 		}
 	}
 	// Configure HTTPS target proxy and forwarding rule, if required.
 	if (ingAnnotations.UseNamedTLS() != "") || len(ing.Spec.TLS) > 0 {
 		// TODO(nikhiljindal): Support creating https target proxy and forwarding rule.
+		fmt.Println("Error: HTTPS not supported yet.")
```
[the remark this answers is not in the document]
Comment (G-Harmon) [reply app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer.go]: done. copied the error msg :)

**Comment by nikhiljindal** [comment]
Thanks for the fixes, @G-Harmon 
I think this is ready to merge.

Needs rebase. Also verify that this passes "make vet" and "make fmt" while you are at it.

**Comment by G-Harmon** [comment]
Rebased and "make vet fmt" passes.

**Comment by nikhiljindal** [comment]
hmm not sure why the tests failed. Link to details fails:
`Unable to load build details from gs://kubernetes-jenkins/pr-logs/pull/GoogleCloudPlatform_k8s-multicluster-ingress/76/pull-kubernetes-multicluster-ingress-test/56`

**Comment by krzyzacy** [comment]
odd, https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-multicluster-ingress-test run 56 is missing

**Comment by BenTheElder** [comment]
This can happen if the job is aborted for some reason.

**Pull request merged**